src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -311,29 +311,15 @@ private static bool ShouldFormatCore(FormattingContext context, TextSpan mapping
             return true;
         }
 
-        var sourceText = context.SourceText;
-        var absoluteIndex = mappingSpan.Start;
-
-        if (mappingSpan.Length > 0)
-        {
-            // Slightly ugly hack to get around the behavior of LocateOwner.
-            // In some cases, using the start of a mapping doesn't work well
-            // because LocateOwner returns the previous node due to it owning the edge.
-            // So, if we can try to find the owner using a position that fully belongs to the current mapping.
-            absoluteIndex = mappingSpan.Start + 1;
-        }
-
-        var change = new SourceChange(absoluteIndex, 0, string.Empty);
         var syntaxTree = context.CodeDocument.GetSyntaxTree();
-        var owner = syntaxTree.Root.LocateOwner(change);
+        var owner = syntaxTree.Root.FindInnermostNode(mappingSpan.Start, includeWhitespace: true);
```

Comment:
@ryzngard I changed this to the overload that just takes a position, to match the old API. You had it as the one that takes a span, and I don't trust that the formatting engine can handle that sort of change, even if it makes sense with the parameters being passed in here.